test/interactions/clickInteractionTests.ts

```diff
@@ -1,158 +1,208 @@
-        svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-        component = new Plottable.Component();
-        component.renderTo(svg);
+  beforeEach(() => {
+    svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+    component = new Plottable.Component();
+    component.renderTo(svg);
 
-        clickInteraction = new Plottable.Interactions.Click();
-        clickInteraction.attachTo(component);
-      });
+    clickInteraction = new Plottable.Interactions.Click();
+    clickInteraction.attachTo(component);
+  });
 
-      it("can register callback using onClick()", () => {
-        let callbackCalled = false;
-        let lastPoint: Plottable.Point;
-        let callback = function(point: Plottable.Point) {
-          callbackCalled = true;
-          lastPoint = point;
-        };
-        clickInteraction.onClick(callback);
-
-        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
-        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
-        assert.isTrue(callbackCalled, "callback called on clicking Component without moving mouse");
-        assert.deepEqual(lastPoint, { x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2 }, "was passed correct point");
-
-        callbackCalled = false;
-        lastPoint = null;
-        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
-        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 4, SVG_HEIGHT / 4);
-        assert.isTrue(callbackCalled, "callback called on clicking and releasing inside the Component");
-        assert.deepEqual(lastPoint, { x: SVG_WIDTH / 4, y: SVG_HEIGHT / 4 }, "was passed mouseup point");
-
-        callbackCalled = false;
-        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
-        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH * 2, SVG_HEIGHT * 2);
-        assert.isFalse(callbackCalled, "callback not called if mouse is released outside Component");
-
-        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH * 2, SVG_HEIGHT * 2);
-        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
-        assert.isFalse(callbackCalled, "callback not called if click was started outside Component");
-
-        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
-        TestMethods.triggerFakeMouseEvent("mousemove", component.content(), SVG_WIDTH * 2, SVG_HEIGHT * 2);
-        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
-        assert.isTrue(callbackCalled, "callback still called if the mouse is moved out then back inside the Component before releasing");
-
-        svg.remove();
-      });
+  afterEach(() => {
+    svg.remove();
+  });
```

Comment:
This is another quirk of Plottable's test infrastructure.  The way that `afterEach()` works is that it runs after each test, regardless of if the test has failed or succeeded.

Thus, if a test were to fail that has an svg element, we would prefer that the svg be on the screen so that we can inspect the DOM, but we cannot use `afterEach()` for this usecase.

As a result, we have `svg.remove()` at the end of every test, instead of having it in `afterEach()`.

That said, @jtlan found a [neat resource](http://stackoverflow.com/a/25173361) recently that might allow us to put this statement in the `afterEach()` but this is newly found and we have not experimented with it yet.